### PR TITLE
Feature metric dashboard part 1 - References

### PR DESCRIPTION
**Pull request by stevehook**
## Context

Product Owners I need to see in real time specific metrics for key features in dashboard so that they can see the impact of changes over time. This PR introduces the feature metrics dashboard and the first set of metrics for references.

## Changes proposed in this pull request

- [x] A new link to _Feature metrics_ from the _Performance_ index page

image

- [x] New feature metric dashboard

image

- [ ] Tidy up front end
- [x] Minor correction to timestamps for references in `TestApplications` (mainly so that requested_at < time in audit trail when referee responds.)

## Guidance to review

- Does the implementation of the query in `FeatureMetrics` (that gets the wait times for reference responses in a given timeframe) make sense? Is there a simpler way? Would it be worthwhile adding an `ApplicationForm#references_complete_at` attribute (or similar) so it's easier to work out the wait times?
- Note that the review app is showing old test data for which `ApplicationReference#requested_at` is wrong (it's after the response) hence the negative numbers for average wait times.
- How should we style this dashboard given that it's going to have several more metrics added after these?

## Link to Trello card

https://trello.com/c/CBmwCoda/2770-feature-metric-dashboard

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by paulrobertlloyd** [comment]
Only comment from me is not sure about using blue blocks for what are essentially headings, not stats. That said, I do like the presentation… I wonder if it’s something more like this:

Screenshot 2021-01-07 at 10 37 52

Have started a discussion on Slack.

File: app/views/support_interface/feature_metrics_dashboard/dashboard.html.erb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<% content_for :title, 'Feature metrics' %>
+
+<h3 class="govuk-heading-m govuk-!-font-size-27 govuk-!-margin-top-8">References</h3>
```
Comment (paulrobertlloyd) [inline app/views/support_interface/feature_metrics_dashboard/dashboard.html.erb:3]: Illogical heading structure, `h2` should follow `h1`…

```suggestion
<h2 class="govuk-heading-m govuk-!-font-size-27 govuk-!-margin-top-8">References</h2>
```

**Comment by stevehook** [comment]
Thanks @paulrobertlloyd, I've updated the design and metrics as per your suggestion and discussion on Slack.

File: app/models/feature_metrics.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
+class FeatureMetrics
+  include ActionView::Helpers::NumberHelper
+
+  def average_time_to_get_references(
+    start_time,
+    end_time = Time.zone.now.beginning_of_day
+  )
+    times_to_get = time_to_get_references(start_time, end_time)
```
Comment (davidgisbey) [inline app/models/feature_metrics.rb:8]: When I read this i wasn't completely sure what times_to_get_references was returning. I'm struggling to come up with a better name though.

`time_to_get_references_for_each_application` seems a bit clunky.

Comment (davidgisbey) [reply app/models/feature_metrics.rb:8]: Actually the above wouldn't really work as it is only returning times where both references are received.

I think it's probs fine as is 👍

**Review by davidgisbey — APPROVED** [review]
Looks great Steve. It's working really nicely locally!

**Review by paulrobertlloyd — APPROVED** [review]
Looks great. We can tweak the design as we add more metrics.

**Pull request merged**